### ISSUE TITLE
Public struct fields vs getters/setters

### ISSUE THREAD

**Issue by IFcoltransG** [juniperparsnips/rink-runtime#12]
@juniperparsnips commented on #8 that it would be useful to decide when to give a struct getter and setter methods for fields, and when to simply make the fields public.

Rink Runtime started as almost a direct translation of C#, so it used almost all of getters and setters. I thought many were unnecessary in the Rust version.

Some pros of getters:
- Easy to add extra checks and logic to the functions
- API never changes
- Internal details can be hidden

Some pros of public fields:
- No boilerplate to write
- Easy to find important methods in impls
- If needs extra logic later, Rust compiler makes it easy to switch API to getters
- Can be restricted to e.g. `pub(crate)`, hiding _some_ internal details

My personal feeling for the cleanest code is to use as few getters and setters as possible, only when needed for doing extra logic, then restrict privacy for the fields only to the scopes they're needed.

**Comment by LeCalicot** [comment]
I would agree with you. Most of the fields are going to be internal to the crate, so there is no point in creating getter/setter everywhere. As you said, only if there is some complicated logic to set the fields it makes sens to have a setter.

**Comment by IFcoltransG** [comment]
It sounds like we should just proceed with mostly public fields for now, then do a pass over them all sometime, where we restrict visibility e.g. `pub(crate)` to only the parts of the codebase where it's needed.
